js/components/select.tsx

```diff
@@ -349,24 +372,30 @@ class Select extends React.Component<IItemsComponentProps, IDropDownState> {
             }
         }
 
-        const selectElement =
-            model.isMultiSelect ?
-                (
-                    <MultiSelect
-                        renderValue={(item) => (
-                            <div className="simple-value">
-                                <span className="item-label">{item.label}</span>
-                                <button className="item-remove" onClick={() => this.props.select(item.value)}>
-                                    <svg class="react-selectize-reset-button" focusable="false" width="8px" height="8px">
-                                        <path d="M0 0 L8 8 M8 0 L 0 8"></path>
-                                    </svg>
-                                </button>
-                            </div>
-                        )}
-                        {...props}
-                    />
-                 ) :
-                <SimpleSelect {...props} />;
+        let selectElement = null;
+
+        if (model.isMultiSelect) {
+            props.dropdownDirection=this.state.dropdownDirection;
+            props.renderValue=(item) => (
+                <div className="simple-value">
+                    <span className="item-label">{item.label}</span>
+                    <button
+                        className="item-remove"
+                        onMouseDown={e => {
+                            this.props.select(item.value);
+                            e.stopPropagation();
+                        }}
+                    >
+                        <svg className="react-selectize-reset-button" focusable="false" width="8px" height="8px">
+                            <path d="M0 0 L8 8 M8 0 L 0 8"></path>
+                        </svg>
+                    </button>
+                </div>
+            );
+            selectElement = <MultiSelect {...props} />
```

Comment:
If this were outside the if block it wouldn't need the else

Reply:
It's in the if block because it's conditional, I don't get what you mean?
Do you mean selectElement should start off as SimpleSelect and then the if re-assigns it to MultiSelect?

Reply:
My bad. I didn't realise one was multiselect and the other singleselect.